commons/all/src/main/java/org/infinispan/commons/util/Version.java

```diff
@@ -148,7 +148,12 @@ public static short getVersionShort(String versionString) {
       if (versionString == null)
          throw new IllegalArgumentException("versionString is null");
 
-      String parts[] = getParts(versionString);
+      String parts[];
+      if (versionString.equals(String.format("${%s}", INFINISPAN_CORE_SCHEMA_VERSION))) {
```

Comment:
I don't see a good reason to introduce this feature @diegolovison . The same is easily achieved by the pattern replacement during maven resource processing at build time, if one absolutely needs this.

Reply:
This comes to my mind now.

We can stop updating all the following files when doing a major release
```
./core/src/test/resources/configs/named-cache-test.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd"
./core/src/test/resources/configs/unified/12.0.xml:3:      xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./core/src/test/resources/configs/config-with-jgroups-stack.xml:3:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/testdriver/junit5/src/test/resources/cache-container/clustered.xml:2:                 xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-fragment-12.0.xsd"
./server/testdriver/junit5/src/test/resources/endpoints/default.xml:2:           xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/testdriver/junit5/src/test/resources/security/none.xml:2:          xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/testdriver/junit5/src/test/resources/jgroups/xsite-stacks.xml:2:         xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-fragment-12.0.xsd
./server/testdriver/junit5/src/test/resources/socket-bindings/default.xml:2:                 xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/testdriver/junit5/src/test/resources/XSiteServerTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/testdriver/junit5/src/test/resources/XSiteServerTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/testdriver/junit5/src/test/resources/interfaces/default.xml:2:        xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/runtime/src/test/resources/configuration/ServerConfigurationParserTest.xml:3:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/runtime/src/test/resources/configuration/ServerConfigurationParserTest.xml:4:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/runtime/src/main/resources/infinispan-defaults.xml:7:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd"
./server/runtime/src/main/server/server/conf/infinispan-xsite.xml:3:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/runtime/src/main/server/server/conf/infinispan-xsite.xml:4:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd
./server/runtime/src/main/server/server/conf/infinispan-local.xml:3:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/runtime/src/main/server/server/conf/infinispan-local.xml:4:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/runtime/src/main/server/server/conf/infinispan.xml:3:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/runtime/src/main/server/server/conf/infinispan.xml:4:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/datasources/default.xml:2:           xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/CustomStoreTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/CustomStoreTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/cache-container/clustered.xml:2:                 xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-fragment-12.0.xsd"
./server/tests/src/test/resources/configuration/cache-container/clustered-authz.xml:2:                 xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-fragment-12.0.xsd"
./server/tests/src/test/resources/configuration/cache-container/clustered-anchored.xml:3:                 xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-fragment-12.0.xsd
./server/tests/src/test/resources/configuration/cache-container/clustered-anchored.xml:4:                                     urn:infinispan:config:anchored:12.0 https://infinispan.org/schemas/infinispan-anchored-config-12.0.xsd"
./server/tests/src/test/resources/configuration/cache-container/default.xml:3:                 xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-fragment-12.0.xsd"
./server/tests/src/test/resources/configuration/endpoints/default-auth-kerberos.xml:2:           xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/endpoints/default-auth-implicit.xml:2:           xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/endpoints/default-auth.xml:2:           xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/endpoints/default-auth-tls.xml:2:           xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/endpoints/default-auth-tls-external.xml:2:           xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/endpoints/default.xml:2:           xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/endpoints/default-oauth.xml:2:           xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/AuthenticationLDAPTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/AuthenticationLDAPTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/AuthenticationServerTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/AuthenticationServerTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/security/keycloak-jwt.xml:2:          xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/security/properties.xml:2:          xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/security/properties-hashed.xml:2:          xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/security/keycloak-oauth2.xml:2:          xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/security/properties-tls.xml:2:          xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/security/ldap.xml:2:          xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/security/ldap-with-kerberos.xml:2:          xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/security/none.xml:2:          xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/security/truststore.xml:2:          xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/jgroups/xsite-stacks.xml:2:         xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-fragment-12.0.xsd
./server/tests/src/test/resources/configuration/jgroups/stacks.xml:2:         xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-fragment-12.0.xsd"
./server/tests/src/test/resources/configuration/socket-bindings/default.xml:2:                 xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/AuthorizationKerberosTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/AuthorizationKerberosTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/AuthorizationPropertiesTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/AuthorizationPropertiesTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/AuthenticationKeyCloakTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/AuthenticationKeyCloakTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/PersistenceTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/PersistenceTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/AuthenticationServerTLSTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/AuthenticationServerTLSTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/ClusteredServerTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/ClusteredServerTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/XSiteServerTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/XSiteServerTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/AuthenticationServerTrustTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/AuthenticationServerTrustTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/AuthenticationServerImplicitTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/AuthenticationServerImplicitTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/AuthorizationLDAPTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/AuthorizationLDAPTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/AuthenticationKerberosTest.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/AuthenticationKerberosTest.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/AnchoredKeys.xml:4:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./server/tests/src/test/resources/configuration/AnchoredKeys.xml:5:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/configuration/interfaces/default.xml:2:        xsi:schemaLocation="urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./server/tests/src/test/resources/cli/pcache.xml:2:            xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd"
./integrationtests/server-integration/server-integration-commons/src/test/resources/infinispan-custom.xml:3:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./integrationtests/server-integration/server-integration-commons/src/test/resources/infinispan-custom.xml:4:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd
./cli/cli-client/src/test/resources/patch/v1/server/conf/infinispan.xml:3:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./cli/cli-client/src/test/resources/patch/v1/server/conf/infinispan.xml:4:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./cli/cli-client/src/test/resources/patch/v2/server/conf/infinispan.xml:3:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./cli/cli-client/src/test/resources/patch/v2/server/conf/infinispan.xml:4:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./cli/cli-client/src/test/resources/patch/v3/server/conf/infinispan.xml:3:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./cli/cli-client/src/test/resources/patch/v3/server/conf/infinispan.xml:4:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
./jcache/tck-runner-remote/src/test/resources/infinispan.xml:3:        xsi:schemaLocation="urn:infinispan:config:12.0 https://infinispan.org/schemas/infinispan-config-12.0.xsd
./jcache/tck-runner-remote/src/test/resources/infinispan.xml:4:                            urn:infinispan:server:12.0 https://infinispan.org/schemas/infinispan-server-12.0.xsd"
```

Reply:
I agree with Adrian, this looks too much like a hack. This POM fragment in Infinispan modules automatically replaces `${infinispan.core.schema.version}` with the current schema version at build time:

```
        <testResources>
            <testResource>
                <directory>${project.basedir}/src/test/resources</directory>
                <filtering>true</filtering>
            </testResource>
        </testResources>
```

One gotcha is that it will also replace other usages of the `${property.name}` syntax, e.g. for the defaults plugin in XSDs  (which need to be excluded) or for system properties without defaults in configuration XMLs (which is why we always use the slightly different `${property.name:default}` in our XMLs). 

We could probably simplify the `maven-resource-plugin` configuration by changing `infinispan-defaults-maven-plugin`'s syntax or by implementing a [custom resource filter](https://maven.apache.org/plugins/maven-resources-plugin/examples/custom-resource-filters.html) that uses a completely different token to represent the version, but I'm not sure it's worth it.

Reply:
> We can stop updating all the following files when doing a major release

I don't think there's anything preventing us from changing all of those files to use `${infinispan.core.schema.version}`, except for convenience (because IntelliJ needs the actual version to validate the XML and offer tab completion).

Reply:
Thanks for the review and tips. Closing